core/src/include/ModelMetadata.hpp

```diff
@@ -117,6 +129,9 @@ class ModelMetadata {
 #ifdef USE_MPI
     const std::string bboxName = "bounding_boxes";
 #endif
+#ifdef USE_XIOS
+    Xios* xiosHandler = nullptr;
```

Comment:
The way I currently have it set up in #751 is that multiple classes (`ParaGridIO` and `ModelMetadata`) have pointers to it. A singleton would make sense in general, yes. It might make test files containing multiple tests that use XIOS a little trickier to set up, but that's not really the important thing. I'll look into how to implement it. Also good point on using smart pointers.